cleanlab_studio/studio/enrichment.py

```diff
@@ -184,38 +192,87 @@ def run(
         )
         return response
 
+    def _get_latest_job(self) -> EnrichmentJob:
+        """Retrieve the latest job and its details."""
+        return self.list_all_jobs()[0]
+
+    def _get_latest_job_status(self) -> dict[str, Any]:
+        """Retrieve the latest job status with details, e.g. processed_rows and average_trustworthiness_score."""
+        latest_job = self._get_latest_job()
+        return api.get_enrichment_job_status(self._api_key, job_id=latest_job["id"])
+
+    def _get_num_processed_rows(self) -> int:
+        num_rows = int(self._get_enrichment_project_dict()["num_rows"])
```

Comment:
if "num_rows" saved in project. I think it's much better to do this on the backend since it would also save a call for UI as well.
For tmw demo, i think this is fine. but can you add a TODO for this. should be a simple iteration 🥇

Reply:
yes, i was planning on moving this logic to the backend instead of client-side, so i'll go ahead and make that change now while it's at the top of my mind